odin/adapters/proxy.py

```diff
@@ -44,35 +47,39 @@ def __init__(self, name, url, request_timeout):
         self.error_string = 'OK'
         self.last_update = 'unknown'
         self.data = {}
+        self.metadata = {}
         self.counter = 0
 
         # Build a parameter tree representation of the proxy target status
         self.status_param_tree = ParameterTree({
-            'url': (self._get_url, None),
-            'status_code': (self._get_status_code, None),
-            'error': (self._get_error_string, None),
-            'last_update': (self._get_last_update, None),
+            'url': (lambda: self.url, None),
+            'status_code': (lambda: self.status_code, None),
+            'error': (lambda: self.error_string, None),
+            'last_update': (lambda: self.last_update, None),
         })
 
         # Build a parameter tree representation of the proxy target data
-        self.data_param_tree = ParameterTree((self._get_data, None))
+        self.data_param_tree = ParameterTree((lambda: self.data, None))
+        self.meta_param_tree = ParameterTree((lambda: self.metadata, None))
 
         # Create an HTTP client instance and set up default request headers
         self.http_client = tornado.httpclient.HTTPClient()
         self.request_headers = {
             'Content-Type': 'application/json',
             'Accept': 'application/json',
         }
-
+        logging.debug("TARGET SET: %s", self.url)
         self.remote_get()  # init the data tree
+        self.remote_get(get_metadata=True)  # init the metadata
 
-    def update(self, request, path):
+    def update(self, request, path, get_metadata=False):
         """
         Update the Proxy Target `ParameterTree` with data from the proxied adapter,
         after issuing a GET or a PUT request to it. It also updates the status code
         and error string if the HTTP request fails.
         """
 
+        # logging.debug("PROXY UPDATE PATH: %s", path)
```

Comment:
Uncomment?

Reply:
If it is uncommented it needs to not be shouty :wink: